hxmath/math/Vector2.hx

```diff
@@ -734,17 +734,17 @@ abstract Vector2(Vector2Type) from Vector2Type to Vector2Type
      * @param pivot     The pivot point to rotate around.
      * @return          The modified object.
      */
-    public inline function rotate(angle:Float, pivot:Vector2):Vector2
+    public inline function rotate(angle:Float, ?pivot:Vector2):Vector2
     {
         var self:Vector2 = this;
         
         var cos = Math.cos(angle);
         var sin = Math.sin(angle);
-        var dx = self.x - pivot.x;
-        var dy = self.y - pivot.y;
+        var dx = pivot == null ? self.x : self.x - pivot.x;
```

Comment:
I suggest turning this into a single if block to reduce branching.

```haxe
var dx:Float = self.x;
var dy:Float = self.y;

if (pivot != null)
{
    dx = self.x - pivot.x;
    dy = self.y - pivot.y;
}
```

Reply:
Will do.

Reply:
Done.